kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraKeyValueTable.java

```diff
@@ -280,62 +348,101 @@ public KeyValueIterator<Bytes, byte[]> all(
 
   @Override
   public BoundStatement insert(
-      final int partitionKey,
+      final int kafkaPartition,
       final Bytes key,
       final byte[] value,
       final long epochMillis
   ) {
+    final int tablePartition = partitioner.tablePartition(kafkaPartition, key);
     return insert
         .bind()
-        .setInt(PARTITION_KEY.bind(), partitionKey)
+        .setInt(PARTITION_KEY.bind(), tablePartition)
         .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()))
         .setInstant(TIMESTAMP.bind(), Instant.ofEpochMilli(epochMillis))
         .setByteBuffer(DATA_VALUE.bind(), ByteBuffer.wrap(value));
   }
 
   @Override
   public BoundStatement delete(
-      final int partitionKey,
+      final int kafkaPartition,
       final Bytes key
   ) {
+    final int tablePartition = partitioner.tablePartition(kafkaPartition, key);
     return delete
         .bind()
-        .setInt(PARTITION_KEY.bind(), partitionKey)
+        .setInt(PARTITION_KEY.bind(), tablePartition)
         .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()));
   }
 
   @Override
-  public MetadataRow metadata(final int partition) {
-    final BoundStatement bound = getMeta
-        .bind()
-        .setInt(PARTITION_KEY.bind(), partition);
-    final List<Row> result = client.execute(bound).all();
+  public long fetchOffset(final int kafkaPartition) {
```

Comment:
thanks for naming the partitions appropriately 🙏 we may even want to (in a follow up) make this type safe by introducing dedicated types (e.g. `TablePartition` and `KafkaPartition`)

Reply:
100% 🚀